morse_code.rb

```diff
@@ -1 +1,57 @@
-
+MORSE_DICT = {
+  'a' => '.-',
+  'b' => '-...',
+  'c' => '-.-.',
+  'd' => '-..',
+  'e' => '.',
+  'f' => '..-.',
+  'g' => '--.',
+  'h' => '....',
+  'i' => '..',
+  'j' => '.---',
+  'k' => '-.-',
+  'l' => '.-..',
+  'm' => '--',
+  'n' => '-.',
+  'o' => '---',
+  'p' => '.--.',
+  'q' => '--.-',
+  'r' => '.-.',
+  's' => '...',
+  't' => '-',
+  'u' => '..-',
+  'v' => '...-',
+  'w' => '.--',
+  'x' => '-..-',
+  'y' => '-.--',
+  'z' => '--..'
+}.freeze
+
+def decode_char(char)
+  print MORSE_DICT.key(char).upcase
+end
+
+decode_char '.-'
+puts ' '
+def decode_word(word)
+  words = word.split
+  words.each do |n|
+    decode_char(n)
+  end
+end
+
+decode_word '-- -.--'
+puts ' '
+
+def decode(sentence)
+  sentences = sentence.split('   ')
+  sentences.each do |n|
+    decode_word(n)
+    print ' '
+  end
+end

```

Comment:
When I test your methods with the following morse code test inputs as per project requirements:

```
p decode_char('.-')
p decode_word('-- -.--')
p decode("-- -.--   -. .- -- .")
p decode(".-   -... --- -..-   ..-. ..- .-.. .-..    --- ..-.    .-. ..- -... .. . ...")
```

The expected output should be:

![image](https://user-images.githubusercontent.com/38088818/191605600-32c8e440-ba55-49f0-8b30-a89a6cad8519.png)

However, this is your output:

![image](https://user-images.githubusercontent.com/38088818/191605899-eb235b84-5e41-4fe3-826a-545a44190d44.png)

- Kindly look into this and make sure that all your methods output the desired outcome.

NB: Even when I run the test code without the `puts` shorthand `p` I still get an undesired output.

![image](https://user-images.githubusercontent.com/38088818/191607083-80b53339-b9f5-4cdc-beb8-c96afc2abac0.png)

Output:

![image](https://user-images.githubusercontent.com/38088818/191607450-32a5db19-a158-47a8-b86a-c2ba263f7bde.png)

Reply:
![ruby](https://user-images.githubusercontent.com/95174521/191611600-86a4539a-e05c-41d4-82cb-2a95746677ec.png)

Reply:
dear @KDlamini  thank you for your good review 
but i dont know what is that happening with you any way i will try to fix it